neo-gui/UI/TransferDialog.cs

```diff
@@ -42,13 +44,31 @@ public Transaction GetTransaction()
             }).ToArray();
             Transaction tx;
             List<TransactionAttribute> attributes = new List<TransactionAttribute>();
+
+            if (comboBox2.SelectedItem == null)
+            {
+                FromAddress = null;
+            }
+            else
+            {
+                FromAddress = ((string)comboBox2.SelectedItem).ToScriptHash();
+            }
+            
             if (cOutputs.Length == 0)
             {
                 tx = new ContractTransaction();
             }
             else
             {
-                UInt160[] addresses = Program.CurrentWallet.GetAccounts().Select(p => p.ScriptHash).ToArray();
+                UInt160[] addresses;
+                if (FromAddress != null)
+                {
+                    addresses = (from add in Program.CurrentWallet.GetAccounts().Select(p => p.ScriptHash).ToArray() where add.Equals(FromAddress) select add).ToArray();
```

Comment:
We usually use the other syntax type
`.Where(u=>u...)`

Reply:
@shargon 
Understand, I will modify them soon.

Thanks